docs/DeveloperGuide.md

```diff
@@ -383,6 +383,38 @@ The sequence diagram is as follows:
 
 The exported file can be subsequently imported back into any other instance of ContaX, similar to the existing `.json` system of import/export.
 
+### Enhanced Find Logic
+
+The previous implementation of the find function is limited to only searching the name field. The enhancement to the feature allows users to search for `Person` objects by more attributes, including by the `phone`, `email` and `address` fields.
+This is done by creating a helper `SearchType` model within the `model.util` package and abstracting the original `NameConstrainsKeywordsPreicate` to form `ConstrainsKeywordsPreicate` then create different inherit of three different types of predicate.
+The sequence diagram is as follows:
+![Enhanced Find Logic](images/FindCommandSequenceDiagram.png)
+
+### Mass Operation Features
+
+#### Chain Command
+
+This extension allows user to chain multiple commands together.
+
+:information_source: **Note:** A command failure may lead to following commands being invalid (e.g. if create person failed, you cannot edit the newly created person), causing an exception to be thrown.
+
+The sequence diagram is as follows:
+![Chain Command](images/ChainCommandSequenceDiagram.png)
+
+#### Range Command
+
+This extension allows user to perform range of commands based on `index`. Hence, `from/INDEX to/INDEX` is essential in parsing the value of input. The validation for `INDEX` followed as original edit command index validation.
```

Comment:
The second sentence is still problematic, I have no idea what you mean to suggest a change. Your latest commit changed the third sentence.

Reply:
can you check the latest one, I tried rephrase it

Reply:
Ok, the new phrasing is understandable.